### ISSUE TITLE
MMM-Fuel showing undfined

### ISSUE THREAD

**Issue by RexFlymo** [fewieden/MMM-Fuel#85]
Platform (Hardware/OS): RPI

Node version: 16.14.2

MagicMirror version: 2.19.0

Module version: 2.3.0

Description of the issue: When Magic Mirror starts module briefly show loading then changes to undefined.

config is as follows

            {
                module: "MMM-Fuel",
                position: "bottom_center",
                config: {
                    api_key: "API Key from API.NSW",
                    provider: "nsw",
                    secret: "API secret from API.NSW",
                    open: false,
                    showOpenOnly: false,
                    lat: -34.5794939,
                    lng: 150.852613,
                    types: ["e5"],
                    sortBy: ["e5"],
                    radius: 5,
                        }
            },

**Comment by fewieden** [comment]
@RexFlymo is there a specific reason that you use v2.3.0 which is still in development (branch develop) since you don't use any of the new features yet?

The last officially released version is v2.2.1 (branch master).

However, I was running both version with your config and got the prices as expected.

Maybe you forgot to update the dependencies switching to v2.3.0.

Do you see any errors in your terminal where you run MagicMirror or in the browser/electron?

### v2.3.0

![develop](https://user-images.githubusercontent.com/9334168/162131500-fbea64eb-fb66-4598-8f53-0db2cdcd2f02.png)

### v2.2.1

![master](https://user-images.githubusercontent.com/9334168/162131563-5676f7be-92d0-4d4b-8ba7-1e0b0345ef64.png)

**Comment by RexFlymo** [comment]
I was using version 2.2.1 and only changed to version 2.3.0 to see if it helped resolve the problem. Not entirely sure how to go back.

I get the same issue if I run MM on my RPI or if I access via HTTP from another machine on the network. It shows the module briefly with a "loading" message but then after a couple of seconds goes to undefined.  

Looking at the pm2 logs there are some errors coming up but I thought they were affecting a different module here they are.

0|MagicMirror  | [07.04.2022 17:32.21.272] [ERROR]
0|MagicMirror  | Whoops! There was an uncaught exception...
0|MagicMirror  | [07.04.2022 17:32.21.307] [ERROR]
0|MagicMirror  | Error: listen EADDRINUSE: address already in use 127.0.0.1:8080
0|MagicMirror  |     at Server.setupListenHandle [as _listen2] (node:net:1334:16)
0|MagicMirror  |     at listenInCluster (node:net:1382:12)
0|MagicMirror  |     at GetAddrInfoReqWrap.doListen [as callback] (node:net:1520:7)
0|MagicMirror  |     at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:73:8) {
0|MagicMirror  |   code: 'EADDRINUSE',
0|MagicMirror  |   errno: -98,
0|MagicMirror  |   syscall: 'listen',
0|MagicMirror  |   address: '127.0.0.1',
0|MagicMirror  |   port: 8080
0|MagicMirror  | }
0|MagicMirror  | [07.04.2022 17:32.21.309] [ERROR] MagicMirror² will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
0|MagicMirror  | [07.04.2022 17:32.21.310] [ERROR]
0|MagicMirror  | If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues



1|mm           | [07.04.2022 17:33.12.328] [ERROR]
1|mm           | Failed to retrieve repo info for MMM-CalendarExt2: Error: Command failed: cd /home/pi/MagicMirror/modules/MMM-CalendarExt2 && git fetch --dry-run
1|mm           | fatal: couldn't find remote ref refs/heads/master

**Comment by RexFlymo** [comment]
After more experimenting I decided that a complete clean install of Raspian and Magic Mirror was called for. I now have this working.

My conclusion is that at some stage I have either removed or missed on of the dependencies. 

Thank you very much for helping me with this as much as you could and for providing what looks to be an awesome module.

Anyone with a similar issue in the future should start by ensuring that they have all the dependencies installed correctly.

Thank you again.

**Comment by fewieden** [comment]
@RexFlymo thank you for your donation.

A general tip for investigating issues with a specific module is to disable all other modules in the config to test the module in isolation, e.g.

```js
{
  module: 'MMM-NFL',
  position: 'top_right',
  disabled: true,
  config: {
    colored: true,
    helmets: false,
    focus_on: ['NYG', 'SF'],
  }
},
```

The error from the logs indicates that there were multiple instances of MagicMirror running. There are commands, e.g. `sudo lsof -i:8080` to check which process is using up the port 8080 and then you can kill them.

```
0|MagicMirror | code: 'EADDRINUSE',
0|MagicMirror | errno: -98,
0|MagicMirror | syscall: 'listen',
0|MagicMirror | address: '127.0.0.1',
0|MagicMirror | port: 8080
```

**Comment by RexFlymo** [comment]
I did think after I posted the logs that running just the one module and getting it working would be better but instead of just disabling the other modules I choose to start with a clean install. Mainly as I believe that due to my inexperience I may have caused issues with some of the changes I had made experimenting with different modules. 

That makes sense and I think I know how I ended up with two instances running. I think that I had created the start script twice once as mm and once as MM. Now that I have started with a clean install I didn't make this error the second time so I am not getting the same errors.
